python/pylibcudf/pylibcudf/io/types.pyx

```diff
@@ -182,6 +182,8 @@ cdef class SourceInfo:
                     raise FileNotFoundError(
                         errno.ENOENT, os.strerror(errno.ENOENT), src
                     )
+                # TODO: Keep the sources alive (self.byte_sources = sources)
+                # for str data (e.g. read_json)?
```

Comment:
I was curious why we need to do this? Should we doing it for other types of sources?

Reply:
I'm guessing that for raw string like data we're passing in views of the data into these readers, and Python will tend to garbage collect the sources sooner than the views that are passed into the readers.

> Should we doing it for other types of sources?

Ideally no, I think generally it's good to not unnecessarily hold a reference to source if not needed so that memory (albeit probably small) is cleaned up